### PR TITLE
Fix: resource quotas

### LINKED ISSUE #378: onboard samples to build specific projects to reduce likelihood of quota errors
In office hours I onboarded data-science-onramp dataproc clusters to the build specific projects to reudce quota errors. we've seen intermittent flaky quota errors here as well, and should onboard it too

(related https://github.com/GoogleCloudPlatform/python-docs-samples/pull/7633)

### PR DESCRIPTION
**Pull request by loferris**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-dataproc/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [N/A] Appropriate docs were updated (if necessary)

Fixes: #378

### CODE REVIEW

**Review by leahecole — APPROVED** [review]

**Comment by leahecole** [comment]
LGTM assuming tests pass

File: samples/snippets/submit_job_to_cluster.py [hunk 1]
```diff
@@ -232,10 +223,8 @@ def main(
         region = get_region_from_zone(zone)
         # Use a regional gRPC endpoint. See:
         # https://cloud.google.com/dataproc/docs/concepts/regional-endpoints
-        client_transport = (
-            cluster_controller_grpc_transport.ClusterControllerGrpcTransport(
-                address="{}-dataproc.googleapis.com:443".format(region)
-            )
+        client_transport = cluster_controller_grpc_transport.ClusterControllerGrpcTransport(
```
[the remark this answers is not in the document]
Comment (loferris) [reply samples/snippets/submit_job_to_cluster.py]: resolved in latest commit!

File: samples/snippets/submit_job_to_cluster.py [hunk 2]
```diff
@@ -232,10 +223,8 @@ def main(
...
+            address="{}-dataproc.googleapis.com:443".format(region)
         )
         job_transport = job_controller_grpc_transport.JobControllerGrpcTransport(
```
[the remark this answers is not in the document]
Comment (loferris) [reply samples/snippets/submit_job_to_cluster.py]: resolved in latest commit!

File: samples/snippets/list_clusters.py [hunk 3]
```diff
@@ -49,10 +42,8 @@ def main(project_id, region):
     else:
         # Use a regional gRPC endpoint. See:
         # https://cloud.google.com/dataproc/docs/concepts/regional-endpoints
-        client_transport = (
-            cluster_controller_grpc_transport.ClusterControllerGrpcTransport(
-                address="{}-dataproc.googleapis.com:443".format(region)
-            )
+        client_transport = cluster_controller_grpc_transport.ClusterControllerGrpcTransport(
```
Comment (bradmiro) [inline samples/snippets/list_clusters.py:45]: Needs API update as well.

**Review by bradmiro — APPROVED** [review]
LGTM pending the one fix in `list_clusters.py`

File: samples/snippets/list_clusters.py (hunk 3 above)
Comment (loferris) [reply samples/snippets/list_clusters.py]: resolved in latest commit!

**Pull request merged**